File: car/src/models/Car.ts
import m from "mongoose";

export interface ICar {
  id: string;
  registrationNumber: string;
  name: string;
  vin: string;
  model: string;
  modelYear: string;
  userId: string;
  chargerId: string;
  batterySize: number;
}

const carSchema = new m.Schema<ICar>({
  registrationNumber: {
    type: String,
    required: true,
  },
  name: {
    type: String,
    required: true,
  },
  vin: {
    type: String,
    required: false,
  },
  model: {
    type: String,
    required: false,
  },
  modelYear: {
    type: String,
    required: false,
  },
  batterySize: {
    type: Number,
    required: false,
  },
  userId: {
    type: String,
    required: false,
  },
  chargerId: {
    type: String,
    required: false,
  },
});

export const Car = m.model<ICar>("Car", carSchema);

carSchema.statics.build = (attrs) => {
  return new Car(attrs);
};

File: Vpp.Cars/Models/Car.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson.Serialization.IdGenerators;

namespace Vpp.Cars.Models;

public class Car
{
    [BsonId(IdGenerator = typeof(CombGuidGenerator))]
    public Guid? Id { get; set; }
    public string? RegistrationNumber { get; set; }
    public string? Name { get; set; }
    public string? Vin { get; set; }
    public string? Model { get; set; }
    public string? ModelYear { get; set; }
    public string? UserId { get; set; }
    public string? ChargerId { get; set; }
    public int? BatterySize { get; set; }
}